broadbean/blueprint.py

```diff
@@ -263,6 +265,73 @@ def description(self):
 
         return desc
 
+    def write_to_json(self, path_to_file: str) -> None:
+        """
+        Writes blueprint to JSON file
+
+        Args:
+            path_to_file: the path to the file to write to ex:
+            path_to_file/blueprint.json
+        """
+        with open(path_to_file, 'w') as fp:
+            json.dump(self.description, fp, indent=4)
+
+    @classmethod
+    def blueprint_from_description(cls, blue_dict):
+        """
+        Returns a blueprint from a description given as a dict
+
+        Args:
+            blue_dict: a dict in the same form as returned by
+            BluePrint.description
+        """
+        knowfunctions = dict([('function PulseAtoms.{}'.format(fun),
```

Comment:
Perhaps `{n: f for (n, f) in PulseAtoms.__dict__.items() if "__" not in n}` is more to the point?

Reply:
I tried but pytest failed

Reply:
Yes, I was much too fast here. I think the way the `BluePrint` interacts with the `PulseAtoms` is ugly enough to convince us to go down a different route for the building block functions (the atoms), but that is for a later discussion.